docs/arch/02-core-concepts.md

```diff
@@ -0,0 +1,693 @@
+
+**Audit events include:**
+- `mcp_tool_call` - Tool executions
+- `mcp_tools_list`, `mcp_resources_list`, `mcp_prompts_list` - List operations
+- `mcp_resource_read` - Resource access
+- `mcp_prompt_get` - Prompt retrieval
+- `mcp_initialize` - Session initialization
+- `sse_connection` - SSE connections
+- `http_request` - Generic HTTP requests
+- And others - see `pkg/audit/mcp_events.go` for complete list of 15 event types
+
+**Event data:**
+- Timestamp, source, outcome
+- Subjects (user, session)
+- Target (endpoint, method, resource)
+- Request/response data (configurable)
+- Duration and metadata
+
+**Implementation:**
+- Audit middleware: `pkg/audit/middleware.go`
+- Event types: `pkg/audit/events.go`
+- Config: `pkg/audit/config.go`
+
+**Related concepts:** Middleware, Authorization, Parse
+
+### Export
+
+**Export** serializes a workload's RunConfig to a portable JSON file.
+
+**Export process:**
+1. Load workload state from disk
+2. Read RunConfig
+3. Serialize to JSON with formatting
+4. Write to file or stdout
+
+**Exported format:**
+- JSON with schema version
+- All configuration fields
+- Permission profile included
+- Middleware configuration included
+
+**Commands:**
+- `thv export <workload> <path>` - Export to file
+
+**Example:** `thv export my-server ./my-server-config.json`
+
+**Implementation:**
+- CLI: `cmd/thv/app/export.go`
+- Serialization: `pkg/runner/config.go`
+
+**Related concepts:** RunConfig, Import, State
+
+### Import
+
+**Import** creates a workload from an exported RunConfig file.
+
+**Import process:**
+1. Read JSON file
+2. Deserialize to RunConfig
+3. Validate schema version
+4. Deploy workload with configuration
+
+**Commands:**
+- `thv run --from-config <file>` - Import and run
+
+**Implementation:**
+- CLI: `cmd/thv/app/run.go`
+- Deserialization: `pkg/runner/config.go`
+
+**Related concepts:** RunConfig, Export, Deploy
+
+### Monitor
+
+**Monitor** watches container health and lifecycle events.
+
+**Monitoring includes:**
+- Container exit detection
+- Health checks (via MCP ping)
+- Automatic proxy shutdown on container exit
+
+**Health checking:**
+- Send MCP `ping` request periodically
+- Check for valid response
+- Shutdown if unhealthy
+
+**Implementation:**
+- Monitor: `pkg/container/monitor.go`
+- Health checker: `pkg/healthcheck/checker.go`
+
+**Related concepts:** Workload, Transport, Proxy
+
+## Relationships
+
+### Workload Composition
+
+```mermaid
+graph TB
+    Workload[Workload]
+    Workload --> RunConfig[RunConfig]
+    Workload --> Runtime[Runtime]
+    Workload --> Transport[Transport]
+    Workload --> State[State]
+
+    RunConfig --> Profile[Permission Profile]
+    RunConfig --> Middleware[Middleware Configs]
+    RunConfig --> EnvVars[Environment Variables]
+
+    Transport --> Proxy[Proxy]
+    Proxy --> Sessions[Sessions]
+
+    style Workload fill:#90caf9
+    style RunConfig fill:#e3f2fd
+    style Transport fill:#81c784
+```
+
+### Request Flow
+
+```mermaid
```

Comment:
The request flow diagram is incomplete. It shows only 4 middleware types but there are 8 middleware types defined in the codebase.

Missing middleware: Token Exchange, Tool Filter, Tool Call Filter, and Telemetry.

Complete flow should include: Auth → [Token Exchange] → [Tool Filter] → [Tool Call Filter] → Parser → [Telemetry] → [Authorization] → [Audit] → Container

(Brackets indicate conditional middleware that are only present if configured)

Evidence: pkg/runner/middleware.go:16-27